src/mm/memory_manager.rs

```diff
@@ -20,46 +33,64 @@ use crate::util::heap::layout::vm_layout_constants::HEAP_START;
 use crate::util::OpaquePointer;
 use crate::vm::VMBinding;
 
-// This file provides a safe Rust API for mmtk-core.
-// We expect the VM binding to inherit and extend this API by:
-// 1. adding their VM-specific functions
-// 2. exposing the functions to native if necessary. And the VM binding needs to manage the unsafety
-//    for exposing this safe API to FFI.
-
-// For example, for mutators, this API provides a Box<Mutator>, and requires a &mut Mutator for allocation.
-// A VM binding can borrow a mutable reference directly from Box<Mutator>, and call alloc(). Alternatively,
-// it can turn the Box pointer to a native pointer (*mut Mutator), then forge a mut reference from the native
-// pointer. In either way, the VM binding code needs to guarantee the safety.
-
-// How the VM gets mutator/collector/tracelocal handles:
-// * Mutator: from bind_mutator() as Box<Mutator>
-// * Collector: from Collection::spawn_worker_thread() as &mut Collector
-// * TraceLocal: Scanning::* as &mut TraceLocal
-
+/// Run the main loop for the GC controller thread. This method does not return.
+///
+/// Arguments:
+/// * `mmtk`: The reference to an MMTk instance.
+/// * `tls`: The thread that will be used as the GC controller.
```

Comment:
`tls` is not clear.
What does thread mean here?
Is it a pointer to a thread-local storage (e.g. an `int`) that my be used by the current function?
Or is it a `pthread_t *` pointing to the thread?

Reply:
It doesnt matter. It is an opaque pointer to mmtk-core. mmtk-core never assumes what it is. It is merely an identifier for a thread.

Reply:
If it is an identifier, it should be constant and unique, which is not clear here.

Reply:
I dont quite get why it is constant. It uniquely identifies an object, which makes it an identifier. A thread pointer can also identify a thread.

Reply:
Though `tls` is confusing, it is used everywhere in the code. I don't suggest change that, unless you plan to change every occurrences of it from the code base - which I believe we dont have time to at the moment.

Reply:
There should be something constant about an identifier (either the pointer address, or the content the pointer points to).
If your name/staff-id changes, it doesn't help identifying you